hlx_statics/scripts/lib-adobeio.js

```diff
@@ -509,4 +509,26 @@ export function decorateAnchorLink(header) {
     '                </svg>';
   header.appendChild(anchorLink);
   // });
+}
+
+/**
+ * Decorates the width of a block
+ * @param {Element} The element to add the width style to.
+ */
+export function decorateWidth(block) {
+  const wid = block.parentElement.parentElement.getAttribute('data-width');
+  if (wid) block.style.width = wid;
+}
+
+/**
+ * Decorates the background color of a block
+ * @param {Element} The element to add the background color style to.
+ */
+export function decorateBackgroundColor(block) {
+  const color = block.parentElement.parentElement.getAttribute('data-backgroundcolor');
+  if (color == "white") {
+    block.parentElement.parentElement.style.backgroundColor = color;
+  } else if (color == "navy") {
+    block.parentElement.parentElement.style.backgroundColor = "rgb(15, 55, 95)";
+  }
```

Comment:
I think these function should be a utility function instead of decorator as they don't decorate anything but rather abstracting the getAttribute method on parent element like a utility would.

Additionally the JSdoc documentation needs to be clearer by mentioning that these functions rely on parentElement attributes and change the attribute of the `block` based on it.

Finally I think we need a fallback incase the block doesn't have the expected parentElement or the attribute you're looking for is missing. With so many blocks utilizing this we might run into some edge cases.

Reply:
Note: these function don't rely on the parent element of the block they're invoked in but rather take width and background color from the metadata block.